stages/org.osbuild.dracut.conf

```diff
@@ -0,0 +1,184 @@
+          "description": "A dracut module, e.g. base, nfs, network ..."
+        }
+      },
+      "add_dracutmodules": {
+        "description": "Additional dracut modules to include.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "A dracut module, e.g. base, nfs, network ..."
+        }
+      },
+      "omit_dracutmodules": {
+        "description": "Dracut modules to not include.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "A dracut module, e.g. base, nfs, network ..."
+        }
+      },
+      "drivers": {
+        "description": "Kernel modules to exclusively include.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "A kernel module without the .ko extension."
+        }
+      },
+      "add_drivers": {
+        "description": "Add a specific kernel modules.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "A kernel module without the .ko extension."
+        }
+      },
+      "force_drivers": {
+        "description": "Add driver and ensure that they are tried to be loaded.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "A kernel module without the .ko extension."
+        }
+      },
+      "filesystems": {
+        "description": "Kernel filesystem modules to exclusively include.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "A kernel module without the .ko extension."
+        }
+      },
+      "install_items": {
+        "description": "Install the specified files.",
+        "type": "array",
+        "items": {
+          "type": "string",
+          "description": "Specify additional files to include in the initramfs."
+        }
+      },
+      "early_microcode": {
+        "description": "Combine early microcode with the initramfs.",
+        "type": "boolean"
+      },
+      "reproducible": {
+        "description": "Create reproducible images.",
+        "type": "boolean"
+      }
+    }
+  }
+}
+"""
+
+
+def bool_to_string(value):
+    return "yes" if value else "no"
+
+
+def create_configuration_file(tree, filename, options):
+    config_files_dir = f"{tree}/usr/lib/dracut/dracut.conf.d"
+
+    # options with format like:
+    # persistent_policy="<policy>"
+    SIMPLE_STR_OPTIONS = [
+        "compress"
+    ]
+    # options with format like:
+    # add_dracutmodules+=" <dracut modules> "
+    ADD_LIST_OPTIONS = [
+        "add_dracutmodules",
+        "dracutmodules",
+        "omit_dracutmodules",
+        "drivers",
+        "add_drivers",
+        "force_drivers",
+        "filesystems",
+        "install_items"
+    ]
+    # options with format like:
+    # reproducible="{yes|no}"
+    BOOL_OPTOINS = [
+        "early_microcode",
+        "reproducible"
+    ]
```

Comment:
I think I would have inverted this and have `option: type`, or even `option: writer_function`, but this is fine too.

Reply:
hmm... I'm not sure I understand what do you mean here. Do you mean that you would extend the schema and specify a type/writer_function for each configuration option?

Reply:
Sorry, was not clear enough
```python
KNOWN_OPTIONS: {
   "compress": "string",
   "add_dracutmodules": "list",
   "early_microcode": "boolean"
}


for option, value in options.items():
   otype = KNOWN_OPTIONS[option]
   if otype == "string":
     f.write(f'{option}="{value}"\n')
   ...

```

or

```python

def string_writer(f, option, value):
    f.write(f'{option}="{value}"\n')

def list_writer(f, option, value):
  ...

KNOWN_OPTIONS: {
   "compress": string_writer,
   "add_dracutmodules": list_writer,
   "early_microcode": bool_writer
}


for option, value in options.items():
   writer = KNOWN_OPTIONS[option]
   writer(f, option, value)
```

The `KNONW_OPTIONS` thing is a bit funny since it basically is already there, in the `SCHEMA`, so we could in theory parse it and use the info form there too.

Reply:
Right. This makes sense and in most cases it will have lower time complexity. I'll rework it...